### DISCUSSION TITLE
Custom assertions not registered on first run

### DISCUSSION THREAD

**Discussion by tstackhouse** [cypress-io/cypress#16037] [Questions and Help]
I'm following the exact example given here: https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/extending-cypress__chai-assertions

And my custom assertion is never running the first time through.  If I run cypress in watch mode and re-run my spec, it works as expected, but if I refresh the browser on that first run it fails again.  It's giving me `assert.foo` is not a function.  The only real difference is I'm adding onto the `assert` function as well, like so:

```
  const assertFoo() =>{/* logic in here*/};

  _chai.Assertion.addMethod('foo', assertFoo);

  _chai.assert.foo = function(exp, act, parent) {
    new _chai.Assertion(exp).to.be.foo(act, parent);
  };
```

Is that possibly the culprit? It's the documented way to add to the `assert.` TDD syntax :/

**Comment by bahmutov** [comment]
Can you place your code that is not running without a reload into a public repo to show the problem?

**Reply by tstackhouse** [reply]
Sure thing!  It's actually pretty reproducible, here's a repo with an example in isolation, checkout the readme for some additional detail: https://github.com/tstackhouse/cypress-assertion
